File: app/src/main/java/com/xpayworld/payment/di/module/ApplicationModule.java
package com.xpayworld.payment.di.module;

import android.app.Application;
import android.content.Context;
import com.xpayworld.payment.di.ApplicationContext;
import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {
    private final Application mApplication;

    public ApplicationModule(Application application) {
        mApplication = application;
    }

    @Provides
    @ApplicationContext
    Context provideContext() {
        return mApplication;
    }

    @Provides
    Application provideApplication() {
        return mApplication;
    }
}

File: app/src/main/java/com/xpayworld/payment/di/module/ApplicationModule.kt
package com.xpayworld.payment.di.module

import android.app.Application
import android.content.Context
import com.xpayworld.payment.di.ApplicationContext
import dagger.Module
import dagger.Provides

@Module
class ApplicationModule(private val mApplication: Application) {

    @Provides
    @ApplicationContext
    internal fun provideContext(): Context {
        return mApplication
    }

    @Provides
    internal fun provideApplication(): Application {
        return mApplication
    }
}
